pyriemann_qiskit/classification.py

```diff
@@ -0,0 +1,360 @@
+
+    WARNING: At the moment this implementation only supports binary
+    classification (eg. Target vs Non-Target experiment)
+
+    Parameters
+    ----------
+    target : int
+        Label of the target symbol
+    q_account_token : string (default:None)
+        If quantum==True and q_account_token provided,
+        the classification task will be running on a IBM quantum backend
+    process_vector : lambda function (default: lambda v: v)
+        Additional processing on the input vectors. eg: downsampling
+    verbose : bool (default:True)
+        If true will output all intermediate results and logs
+    quantum : bool (default: True)
+        - If true will run on local or remote backend
+        (depending on q_account_token value).
+        - If false, will perform classical computing instead
+    test_input : dict
+        Contains vectorized test set for target and non-target classes
+
+    Notes
+    -----
+    .. versionadded:: 0.0.1
+
+    Attributes
+    ----------
+    classes_ : list
+        list of classes.
+
+    See Also
+    --------
+    QuanticSVM
+    QuanticVQC
+
+    """
+
+    def __init__(self, target, q_account_token=None, quantum=True,
+                 process_vector=lambda v: v, verbose=True, test_input={}):
+        self.verbose = verbose
+        self._log("Initializing Quantum Classifier")
+        self.test_input = test_input
+        self.process_vector = process_vector
+        self.q_account_token = q_account_token
+        self.target = target
+        self.quantum = quantum
+        # protected field for child classes
+        self._training_input = {}
+
+    def _init_quantum(self):
+        if self.quantum:
+            aqua_globals.random_seed = datetime.now().microsecond
+            self._log("seed = ", aqua_globals.random_seed)
+            if self.q_account_token:
+                self._log("Real quantum computation will be performed")
+                IBMQ.delete_account()
+                IBMQ.save_account(self.q_account_token)
+                IBMQ.load_account()
+                self._log("Getting provider...")
+                self._provider = IBMQ.get_provider(hub='ibm-q')
+            else:
+                self._log("Quantum simulation will be performed")
+                self._backend = BasicAer.get_backend('qasm_simulator')
+        else:
+            self._log("Classical SVM will be performed")
+
+    def _log(self, *values):
+        if self.verbose:
+            print("[QClass] ", *values)
+
+    def _vectorize(self, X):
+        vector = X.reshape(len(X), self._feature_dim)
+        return [self.process_vector(x) for x in vector]
+
+    def _split_target_and_non_target(self, X, y):
+        self._log("""[Warning] Spitting target from non target.
+                 Only binary classification is supported.""")
+        n_matrices = len(X[0])
+        try:
+            n_channels = len(X[0][0])
+        except Exception:
+            n_channels = 1
```

Comment:
You should test the suggestion before acceptation.
```suggestion
        X = np.atleast_2d(X)
        n_matrices, n_channels = X.shape
```

Reply:
Nice! X being a set of matrices, `X = np.atleast_2d(X[0])` should works.